### PR TITLE
Remove obsolete setting link preview

### PR DESCRIPTION
**Pull request by timkrueger**
The setting is obsolete since the functionality for media detection in
text messages was removed.

See: commit 8468535a

Signed-off-by: Tim Krüger <t@timkrueger.me>

### CODE REVIEW

**Comment by nickvergessen** [comment]
Do we need an update to remove the setting from a database or something? So in case we ever use it again some people don't have it preset

**Comment by timkrueger** [comment]
> Do we need an update to remove the setting from a database or something? So in case we ever use it again some people don't have it preset

Good you say it. I didn't thought about it. By my feeling it should be removed. 

Is this a manual step or is already some :mage: will do it?

//cc @mahibi  @AndyScherzinger

**Comment by AndyScherzinger** [comment]
> Is this a manual step or is already some 🧙 will do it?

No magic, you'd have to manually remove it.

**Comment by AndyScherzinger** [comment]
@timkrueger translations are handled by the transifex bot and will be overridden, so you only need to remove the strings from the master file `values/strings.xml`

**Comment by timkrueger** [comment]
I think I destroyed something during git rebase. I accidentally included commits from the master. 

I'll create a new branch and make a new PR.

**Review by AndyScherzinger — APPROVED** [review]
LGTM while it be nice to have a TODO comment in the code to remove it with 13.0.0 (for example)

File: app/src/main/java/com/nextcloud/talk/utils/preferences/AppPreferences.java [hunk 1]
```diff
@@ -253,9 +251,7 @@
     @KeyByString("phone_book_integration") 
     void setPhoneBookIntegration(boolean value);
 
-    @KeyByString("link_previews")
-    void setLinkPreviewsAllowed(boolean value);
-
+    // TODO: Remove in 13.0.0
```
Comment (AndyScherzinger) [inline app/src/main/java/com/nextcloud/talk/utils/preferences/AppPreferences.java:254]: ❤️

**Comment by timkrueger** [comment]
Backported manually in #1484

**Pull request merged**